zerver/models.py

```diff
@@ -1652,6 +1652,7 @@ class AbstractMessage(models.Model):
     has_attachment: bool = models.BooleanField(default=False, db_index=True)
     has_image: bool = models.BooleanField(default=False, db_index=True)
     has_link: bool = models.BooleanField(default=False, db_index=True)
+    has_reaction: bool = models.BooleanField(default=False, db_index = True)
```

Comment:
Do we need the field, or would it be efficient to do a query on the Reaction table to do this check?

Reply:
@timabbott I was wondering the same thing when I started working on this. I decided to go in this direction because it was a solution that required almost no code changes, as it took advantage of all the logic of the other 'has' attributes.

Should I change my strategy run the query on the reaction table?

Reply:
I can also build the query with the Reaction table and run it using the EXPLAIN and bring the results to this thread so we can check if it's efficient enough for this use case.

Reply:
I would do a quick prototype of what the Reaction table query would be (without updating this PR) and just post the query here; I can test in on a production system to see if we can optimize it appropriately.  If the answer is no, we should go with this approach.

Reply:
Well, after a bit of testing, I think we basically do need this field to make this query efficient.  I'll post notes on what needs to happen for this to actually work.